api/observability/v1/output_types.go

```diff
@@ -59,10 +59,22 @@ var (
 )
 
 // OutputSpec defines a destination for log messages.
+// +kubebuilder:validation:XValidation:rule="self.type != 'azureMonitor' || has(self.azureMonitor)", message="Additional type specific spec is required for the output type"
```

Comment:
nit: Maybe make sense to add output type name to the validation message to be more clear